Stefan jum/add dev content

StefanJum opened the pull request.

StefanJum committed:
Use `workdir/` as clone directory instead of `.unikraft/`

`.unikraft/` was previously used as the directory to clone library
 repositories (including the `unikraft` core repository). Being a
hidden directory, it is not immediately available to the developer.

Use `workdir/` instead.
Also use `workdir/build/` as output directory and ignore the `workdir/`
directory.

Signed-off-by: Stefan Jumarea <stefanjumarea02@gmail.com>

StefanJum committed:
Move config files to defconfigs/ directory

Move all the defconfig files into the `defconfigs/` directory.
Also minimise them using `make savedefconfig`, removing all
unnecesary config options.

Signed-off-by: Stefan Jumarea <stefanjumarea02@gmail.com>

StefanJum committed:
Add run config yaml file and setup script

The `generate.py` script will be downloaded from
https://raw.githubusercontent.com/unikraft/app-testing/staging/scripts/generate.py
and used to generate the scripts in the `scripts/build/` and `scripts/run/`
directories. This will improve maintenance (only required to happen for the
`generate.py` script), while still keeping things usable by providing the
basic build and run scripts.

Use the `setup.sh` script will clone all the dependencies in the
`workdir` repository.

Signed-off-by: Stefan Jumarea <stefanjumarea02@gmail.com>

razvand commented:
Please also add an update to the `README.md` file. See the updates from [this commit](https://github.com/unikraft/app-elfloader/commit/450a52101b17ed041ae97b9d219c997669309cdf).

StefanJum committed:
README.md: Add docs on using the generate scripts

Add instructions on using the generate script and the scripts that will
result from that for building and running the application.

Signed-off-by: Stefan Jumarea <stefanjumarea02@gmail.com>

StefanJum commented:
Done @razvand

razvand commented on README.md:
```suggestion
To make it easier to build, run and test different configurations, the repository provides a set of scripts that do everything required.
```

razvand commented on README.md:
```suggestion
First of all, grab the [`generate.py` script](https://github.com/unikraft/app-testing/blob/staging/scripts/generate.py) and place it in the `scripts/` directory by running:

'''console
wget https://raw.githubusercontent.com/unikraft/app-testing/staging/scripts/generate.py -O scripts/generate.py
chmod a+x scripts/generate.py
'''

Now, run the `generate.py` script.
You must run it in the root directory of this repository:
```

Replace the apostrophes above with backticks.

razvand reviewed (changes requested):
Remove the `generate.py` script, since it's now downloaded. Also ignore it.

razvand commented:
Also update the TLDR with the downloading and using of `generate.py`, same as documented [here](https://github.com/unikraft/app-llama2-c/#quick-set-up-aka-tldr).

StefanJum committed:
.gitignore: Ignore all config output files

Signed-off-by: Stefan Jumarea <stefanjumarea02@gmail.com>

StefanJum committed:
kraft.yaml: Support only QEMU and Firecracker targets

Only keep `qemu` and `firecracker` targets in `kraft.yaml`.

Signed-off-by: Stefan Jumarea <stefanjumarea02@gmail.com>

razvand reviewed (approved):
Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>
Approved-by: Razvan Deaconescu <razvand@unikraft.io>
